src/asm_files.cpp

```diff
@@ -47,6 +48,27 @@ EbpfMapDescriptor* find_map_descriptor(int map_fd) {
     return nullptr;
 }
 
+// parse_maps_sections processes all maps sections in the provided ELF file by calling the platform-specific maps
+// parser. The section index of each maps section is inserted into section_indices. Maps sections are identified as any
+// section called "maps", or matching "maps/<map-name>".
+void parse_map_sections(const ebpf_verifier_options_t* options, const ebpf_platform_t* platform, const ELFIO::elfio& reader, std::vector<EbpfMapDescriptor>& map_descriptors, std::set<ELFIO::Elf_Half>& section_indices) {
+    ELFIO::section* s = reader.sections["maps"];
+    if (s) {
+        platform->parse_maps_section(map_descriptors, s->get_data(), s->get_size(), platform, *options);
+        section_indices.insert(s->get_index());
+    }
+
+    std::string maps_prefix = "maps/";
+    for (ELFIO::Elf_Half i = 0; i < reader.sections.size(); ++i) {
+        s = reader.sections[i];
+        auto name = s->get_name();
+        if (name.length() > 5 && name.compare(0, maps_prefix.length(), maps_prefix) == 0) {
+            platform->parse_maps_section(map_descriptors, s->get_data(), s->get_size(), platform, *options);
+            section_indices.insert(s->get_index());
+        }
+    }
+}
```

Comment:
I think it's better to merge both branches, so a valid name has either `'\0'` or `'/'` in the 5th index.

Also, it should be then possible to encapsulate the decision in a dedicated function `is_map_section`, accepting a string and returning a boolean.

Reply:
Done